src/polygon.jl

```diff
@@ -0,0 +1,508 @@
+    makechildpopulation(population1::Array, population2::Array, number_triangles::Int)
+
+Takes 2 images and returns a new image with triangles randomly taken from both images.
+The order of the triangles is retained since it plays a big role in how the image looks.
+Only the triangle list is updated, not yet the image or score.
+"""
+function makechildpopulation(population1::Array, population2::Array, number_triangles::Int)
+    childTriangles = Vector{Triangle}(undef, number_triangles)
+    for i in 1:number_triangles
+        if rand() > 0.5
+            childTriangles[i] = population1[1][i]
+        else
+            childTriangles[i] = population2[1][i]
+        end
+    end
+
+    child = [childTriangles, population1[2], population1[3]]
+    # The image and score aren't updated yet but that's alright since it will happen in the next step of the algorithm
+
+    return child
+
+end
+
+
+"""
+    triangleevolution(image::String, number_triangles::Int, generations::Int)
+
+An evolutionary algorithm with the goal to approximate a given image as well as possible with a bunch of triangles.
+If a name for a gif is given (e.g. gifname = "nicegif.gif"), it will also make a gif out of the best individual's image for every generation.
+There's a lot of parameters here to play around with, try out some crazy stuff!
+Inputs:
+
+REQUIRED
+- image: The name of your target image
+
+OPTIONAL
+- number_triangles: The amount of triangles the algorithm uses to try and recreate your image
+- generations: The amount of generations in the evolutionary algorithm
+- pop_size: The population size per generation
+- elitism_freq: The fraction of your population that gets to be part of the elite. The best individuals of your generation will become part of the elite and get a free pass to the next generation.
+- newblood_freq: The fraction of your population that gets replaced by new, random individuals every generation.
+- mutation_freq: The chance for a triangle (= "a gene") of an individual to get mutated every generation
+- gifname: If you want to make a gif out of the evolutionary proces, enter a gifname like "amazinggif.gif"
+- fps: The FPS of your gif
+"""
+function triangleevolution(image::String = "src/figures/TotoroTester.jpeg"; number_triangles::Int = 25, generations::Int = 50, pop_size::Int = 70, elitism_freq::Number = 0.10, newblood_freq::Number =  0.03, mutation_freq::Number = 0.07, gifname = nothing, fps = 2)
+    
+    makegif = !isnothing(gifname) # Only make the result into a gif if the user enters a name for it
+
+    elite_size = Int(round(pop_size*elitism_freq)) # The elite of every generation (best individuals) will get a free pass to the next generation
+    newblood_size = Int(round(pop_size*newblood_freq)) # A part of the population will be replaced with new, random individuals every generation.
+    # This keeps our gene pool fresh and sparkly!
+
+    if image[1:5] == "https" #file from internet: credits to Dan Getz from stackoverflow for this
```

Comment:
I wish people would credit my StackOverflow answers in their codebases :grinning: